tests/controllers/settings_info_ctrl_spec.coffee

```diff
@@ -1,57 +1,60 @@
 describe "SettingsInfoCtrl", ->
   scope = undefined
   Wallet = undefined
-  pairingCode = undefined
+  Alerts = undefined
+  $q = undefined
 
   beforeEach angular.mock.module("walletApp")
 
   beforeEach ->
-    angular.mock.inject ($injector, $rootScope, $controller) ->
+    angular.mock.inject ($injector, $rootScope, $controller, _$q_) ->
       Wallet = $injector.get("Wallet")
-      MyWallet = $injector.get("MyWallet")
+      Alerts = $injector.get("Alerts")
+      $q = _$q_
 
-      Wallet.status.isLoggedIn = true
+      Wallet.user.guid = "user_guid"
+      Wallet.user.alias = "user_alias"
       Wallet.makePairingCode = (success, error) ->
-        if scope.pairingCode
-          success()
-        else
-          error()
+        if scope.pairingCode then error() else success("code")
+      Wallet.removeAlias = () -> $q.resolve()
 
       scope = $rootScope.$new()
 
       $controller "SettingsInfoCtrl",
-        $scope: scope,
-        $stateParams: {},
+        $scope: scope
 
       scope.$digest()
 
-      return
+  describe "remove alias", ->
+    it "should ask to confirm", ->
+      spyOn(Alerts, 'confirm').and.callThrough()
+      scope.removeAlias()
+      expect(Alerts.confirm).toHaveBeenCalledWith(jasmine.any(String), { id: 'user_guid' })
 
-    return
-
-  describe "show pairing code", ->
-    it "should show pairing code if valid", ->
-      success = () ->
-      error = () ->
+    it "should remove the alias", ->
+      spyOn(Alerts, 'confirm').and.returnValue($q.resolve())
+      spyOn(Wallet, 'removeAlias').and.returnValue($q.resolve())
+      scope.removeAlias()
+      scope.$digest()
+      expect(Wallet.removeAlias).toHaveBeenCalled()
+      expect(scope.loading.alias).toEqual(false)
 
-      scope.pairingCode = 'code'
+  describe "pairing code", ->
+    it "should show if valid", ->
       scope.showPairingCode()
-      Wallet.makePairingCode(success, error)
-      expect(scope.display.pairingCode).toBe(true)
+      scope.$digest()
+      expect(scope.pairingCode).toEqual("code")
+      expect(scope.loading.code).toEqual(false)
 
-    it "should show an error when failed", inject((Alerts) ->
-      success = () ->
-      error = () ->
+    it "should display an error when show failed", ->
```

Comment:
Maybe change to "should display an error _and disable spinner_ when show failed"